packages/consensus/src/types/utxo_set.cairo

```diff
@@ -50,3 +52,92 @@ pub impl UtxoSetImpl of UtxoSetTrait {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
Can you please also add a script/notebook/snippet (e.g. in python) to generate test data & do cross-checks? (you'd need it anyways)

Reply:
I need more informations, a python script to generate an array of outpoints? so RPC query on the outputs of a block, then build the associated outpoint to add it in the utreexo accumulator?

And to check:
Do I make a python script separate from the program, where the input data is hard coded, to serve as a reference to check that a unit test of leaves_to_add: Array<felt252> is correct?

Or integrate the script into the integration tests? but in this case there's the problem of retrieving the input (hashed array of outpoints) for the py script directly from the cairo code. 

Or create a sh script in consensus/scarb.toml like:
utreexo_test = “scarb build && utreexo_unitaire_tests.sh”

sh:
// generate array<OutPoint>
arguments=$(python ../../scripts/data/utreexo_outpoints_args.py ${block_height})
// Compute utreexo from cairo
output=$(scarb cairo-run --no-build --function test_utreexo_add “$arguments”)
// Compute utreexo from python
ref_output = $(python ../../scripts/data/compute_utreexo.py ${arguments}

output != ref_output ?

Reply:
@Jeanmichel7 for now you can use the utreexo.py (that @bloomingpeach introduced) to generate data for your unit tests and save the snippet (specific to your tests) somewhere. For instance put it in the comments.

The goal is to:
- have a reference code that implements the same algo
- have a reproducible way to generate test data